.github/workflows/test.yml

```diff
@@ -18,12 +18,93 @@ concurrency:
   cancel-in-progress: true
 
 env:
-  RUSTFLAGS: '--cfg async_executor_impl="async-std" --cfg async_channel_impl="async-std"'
+  RUSTFLAGS: '--cfg async_executor_impl="async-std" --cfg async_channel_impl="async-std" --cfg hotshot_example'
   RUST_LOG: info,libp2p=off,node=error
   CARGO_TERM_COLOR: always
 
 jobs:
+  build-test-artifacts:
+    name: Build test artifacts
+    runs-on: ubuntu-latest
+    steps:
+      - uses: rui314/setup-mold@v1
+
+      - uses: actions/checkout@v4
+        with:
+          # By default actions/checkout checks out a merge commit. Check out the PR head instead.
+          # https://github.com/actions/checkout#checkout-pull-request-head-commit-instead-of-merge-commit
+          ref: ${{ github.event.pull_request.head.sha }}
+          submodules: recursive
+      - name: Install nextest
+        uses: taiki-e/install-action@nextest
+
+      - name: Enable Rust Caching
+        uses: Swatinem/rust-cache@v2
+        with:
+          cache-on-failure: true
```

Comment:
Is this a good idea? Doesn't it mean that if we have a failure we only get a partial cache and if we afterwards get a successful build for the same cache key we don't save it and keep a partial cache?

Reply:
I don't understand why it would prevent the cache from being saved on a successful run. But I'll remove it isn't adding much any way.

Reply:
fbd508ef5b21976bce3edcd32100a35a58f65640